src/v/cluster/archival_metadata_stm.cc

```diff
@@ -1189,7 +1205,18 @@ void archival_metadata_stm::apply_add_segment(const segment& segment) {
     bool disable_safe_add
       = config::shard_local_cfg()
           .cloud_storage_disable_metadata_consistency_checks.value();
-    if (!disable_safe_add && !_manifest->safe_segment_meta_to_add(meta)) {
+    if (
+      !disable_safe_add && segment.is_validated == segment_validated::yes
+      && !_manifest->safe_segment_meta_to_add(meta)) {
```

Comment:
Does this mean that we can now set `cloud_storage_disable_metadata_consistency_checks` back to `false`?

Reply:
are you tracking that @Lazin or @VladLazar ?

Reply:
Created a follow up issue: https://github.com/redpanda-data/redpanda/issues/14824